matrix_messages.rst

```diff
@@ -0,0 +1,92 @@
+Raiden transport messages
+#########################
+
+Definitions
+===========
+* Sender: entity sending the message
+* Recipient: entity able of receiving and decoding the message
+* Message: A data object exchanged between Sender and Recipient
+* Message Field: an attribute of JSON data format that is required to be included in the message
+
+Transport envelope
+==================
+Messages are exchanged in JSON format.
+Message itself contains metadata that allow recipient to verify sender of the message (via use of PK signatures).
+
+Rationale
+---------
+* plaintext JSON messages are easy to read for both humans and machines
+* JSON format is widely used and can be easily validated using JSON-schema (http://json-schema.org/)
+* JSON is easily extensible and extension can be made backward-compatible
+* Message format must not be dependent on the transport used - it should not matter whether the message is sent via Matrix or REST
+* Sender identity verification must not be part of the transport. It should be possible to submit the message over an untrusted channel or delegate.
+
+Example of a message
+-------------------
+``
+{
+    'signature': '0xabcd....01234',
+    'content-type’: 'base-64',
+    'data': 'eyAnaGVhZGVyJzogeyAndHlwZScgOiAndGV4dC1tZXNzYWdlJywgJ3NlbmRlcic6ICAnMHhhYWVlLi4uMTIzNCcsICd0aW1lc3RhbXAnOiAxMjM0NTY3ODkgfSwNCidib2R5JzogeyAnbWVzc2FnZSc6ICdoZWxsbyB3b3JsZCcgfX0='
+}
+``
+
+Envelope fields:
+===============
+* signature - a sender’s signature of the data field
+* content-type - how the data are encoded, i.e.:
+    * text-plain: a simple escaped JSON stored as a JSON string
+    * base-64: base-64 encoded JSON
+* data - tranmitted data
+Message MUST contain all mentioned fields and MUST be discarded by the Recipient otherwise.
+
+After unpacking `data` using the method(s) specified in content-type field, recipient MUST obtain a JSON object of a following format:
+``
+{
+    'header': {
+        'sender': '',
+        'timestamp': '',
+        'type': ''
+    }
+    'body': { <JSON OBJECT> }
+}
+``
+
+Header fields
+=============
+* Sender (string) - public key used to verify signature of the message. Recipient MUST validate the signature and MUST discard the message if the signature doesn’t match
+* Timestamp (int) - time the message was sent (note: a block hash might also be used to enforce timestamp tampering)
+* Type (string) - user-specified type of the message
```

Comment:
Is there a list of the possible types?

Reply:
not yet, but we can reuse what's in Raiden already:

```
    'Ack',
    'Ping',
    'SecretRequest',
    'Secret',
    'DirectTransfer',
    'Lock',
    'LockedTransfer',
    'MediatedTransfer',
    'RefundTransfer',
```